features/ui/src/main/java/org/opennms/alec/rest/ALECRestImpl.java

```diff
@@ -1,26 +1,119 @@
 package org.opennms.alec.rest;
 
+import java.util.Arrays;
+import java.util.Optional;
+
 import javax.ws.rs.core.Response;
 
+import org.opennms.alec.data.DataStore;
+import org.opennms.alec.driver.main.Driver;
+import org.opennms.alec.engine.api.EngineFactory;
+import org.opennms.alec.engine.api.EngineRegistry;
+import org.opennms.alec.engine.cluster.ClusterEngineFactory;
+import org.opennms.alec.engine.dbscan.DBScanEngineFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceReference;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import com.fasterxml.jackson.databind.ObjectMapper;
+
 public class ALECRestImpl implements ALECRest {
     private static final Logger LOG = LoggerFactory.getLogger(ALECRestImpl.class);
 
+    private final DataStore dataStore;
+    private final BundleContext bundleContext;
+
+    public ALECRestImpl(DataStore dataStore, BundleContext bundleContext) {
+        this.dataStore = dataStore;
+        this.bundleContext = bundleContext;
+    }
+
     @Override
     public Response ping() {
-        // How do I identify the user?
-
+        return Response.ok("pong!!").build();
+    }
 
+    @Override
+    public Response getDB(String key) {
```

Comment:
I think we should move away defining a generic key/value store here in the REST API and use concrete objects intead.

Reply:
I refactored the api